File: java/org.financer.client/src/main/java/org/financer/client/local/Application.java
package org.financer.client.local;

import java.util.List;

public interface Application {
    public enum MessageType {
        ERROR("error"),
        INFO("info"),
        WARNING("warning"),
        SUCCESS("success");

        private final String name;

        MessageType(String name) {
            this.name = name;
        }

        public String getName() {
            return name;
        }
    }

    void showLoadingBox();

    void hideLoadingBox();

    void setOffline();

    void setOnline();

    void showToast(MessageType messageType, String message);

    default void showToast(MessageType messageType, List<String> messages) {
        messages.forEach(message -> showToast(messageType, message));
    }

    void showErrorDialog(Exception exception);
}

File: java/org.financer.client/src/main/kotlin/org/financer/client/local/Application.kt
package org.financer.client.local

import java.util.function.Consumer

interface Application {
    enum class MessageType(val type: String) {
        ERROR("error"),
        INFO("info"),
        WARNING("warning"),
        SUCCESS("success");
    }

    fun showLoadingBox()
    fun hideLoadingBox()
    fun setOffline()
    fun setOnline()
    fun showToast(messageType: MessageType?, message: String?)
    fun showToast(messageType: MessageType?, messages: List<String?>?) {
        messages!!.forEach(Consumer { message: String? -> showToast(messageType, message) })
    }

    fun showErrorDialog(exception: Exception?)
}